packages/integration-tests/src/entities/Author.ts

```diff
@@ -89,7 +89,7 @@ authorConfig.addRule((a) => {
 });
 
 authorConfig.addRule((a) => {
-  if (a.changes.lastName.hasChanged) {
+  if (!a.isNewEntity && a.changes.lastName.hasChanged) {
```

Comment:
While I approve of the changes to this PR, I do worry that bringing this version of joist into our code base may result in hard to find bugs where the old behviour was assumed. 🤔

Reply:
Yeah, we do have ~90 calls to `hasChanged` currently in the codebase, but probably still doable to audit + run the tests and :crossed_fingers: should be good enough. If we miss one or two I think it's worth the long-term UX win.